scripts/buf_lint_check.sh

```diff
@@ -32,6 +32,8 @@ populate_jar_config_file_paths() {
 
   if [ "$1" = "Linux" ]; then
     buf_file_name="buf-Linux-x86_64"
+  elif [ "$1" = "Darwin" ] && [ "$(uname -m)" = "arm64" ]; then
+    buf_file_name="buf-Darwin-arm64"
   else
     buf_file_name="buf-Darwin-x86_64"
   fi
```

Comment:
```suggestion
case "$1" in
  Linux)
    buf_file_name="buf-Linux-x86_64"
    ;;
  Darwin)
    if [ "$(uname -m)" = "arm64" ]; then
      buf_file_name="buf-Darwin-arm64"
    else
      buf_file_name="buf-Darwin-x86_64"
    fi
    ;;
  *)
    echo "Protobuf lint check not available on $1"
    exit 0
    ;;
esac
```
 Nit: we could use `case` block instead of `if/elif` block would be more cleaner.